Controller/AuthController.php

```diff
@@ -159,7 +159,7 @@ public function syncWallets(Request $request)
                 try {
 
                     // Check if wallet already exist
-                    $wallet = $walletRepository->findOneBy(['user' => $userObj->getId(), 'address' => $account['address']]); // ,
+                    $wallet = $walletRepository->findOneBy(['user' => $userObj->getId(), 'address' => $account['address'], 'network' => $account['network']]);
```

Comment:
This will result in false negatives due to address normarization I mentioned earlier. Replace `'address' => $account['address']` above with `'path' => $account['path']`

Reply:
Updated.